src/scheduler/node_info.c

```diff
@@ -4933,12 +4933,7 @@ parse_execvnode(char *execvnode, server_info *sinfo)
 		return NULL;
 	}
 
-	excvndup = strdup(execvnode);
-	if (excvndup == NULL) {
-		free(nspec_arr);
-		log_err(errno, __func__, MEM_ERR_MSG);
-		return NULL;
-	}
+	excvndup = string_dup(execvnode);
```

Comment:
string_dup() can still return NULL on a failed malloc().  It just handles the logging for you.